### PR TITLE
feat: add explainer page

### LINKED ISSUE #41: Give a cool explainer per example when people hover over a questionmark
It would be super cool, if there is a good explainer per example, when people hover over a symbol, like a question mark. 

For example, it is not clear to most users why our Fibonacci example starts with `[1, 1]` or why the game of life inputs and outputs are as they are.

So it can be a simple site or pop-up that explains one example step by step with graphics.

### PR DESCRIPTION
**Pull request by Dominik1999**
Adding an explainer page in markdown. The user can click the question mark symbol in the `CodingEnvironment`. 

We explain on the page what the playground is and how to use it. I am sure we will have several iterations about the explaining page and also the positioning of the question mark symbol. 

closes #41

### CODE REVIEW

File: playground/src/markdown/Explainer.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+![](https://i.imgur.com/y3yYt2R.png)
+
+The Miden VM works like a regular virtual machine. Only that you can provide secret inputs (Witness) and together with the final state the Miden VM creates a proof for it. A zero-knowledge virtual machines proves that a given set of inputs and program code results in the final state.  
+
+![](https://i.imgur.com/t517366.png)
+
+## Inputs of the Miden VM
+External inputs can be provided to Miden VM in the Playground in two ways:
+
+![](https://i.imgur.com/n0LZpvU.png)
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:17]: This image isn't great

image

The gradient on the left side doesn't seem to have any purpose and the labels are not really aligned. Perhaps remove the image and use inline code that is native to common markdown?

```suggestion
```json
{
  "stack_init": ["0"],
  "advice_tape": ["0"]
}
```

Even better if we can add syntax highlight

File: playground/src/markdown/Explainer.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+To get to know the language, check out our examples. It's better than writing in Circuits ...
+
+![](https://i.imgur.com/tyeMjD9.png)
+
+Want to know more? [Here](https://wiki.polygon.technology/docs/miden/user_docs/assembly/main).
+
+## Outputs of the Miden VM
+
+![](https://i.imgur.com/Bjgky7b.png)
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:41]: Here too I would replace it with an inline json and move the labels in the image to text (in general I'd avoid doing this, since search engines and blind people are excluded from reading it)

File: playground/src/markdown/Explainer.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# What the f*ck? A short explainer
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply playground/src/markdown/Explainer.md:1]: It seems none of the headers have anchors, that makes sharing documentation harder. It would be nice to have a table of contents and anchors like we have in the mdbook

File: playground/src/markdown/Explainer.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+Want to know more? [Here](https://wiki.polygon.technology/docs/miden/intro/overview#inputs-and-outputs).
+
+## Program Code
+![](https://i.imgur.com/xb7Iodq.png)
+
+Our goal is to make Miden VM an easy compilation target for high-level blockchain-centric languages such as Move and Solidity.  However, compilers to help with this have not been developed yet. Thus, for now, the primary way to write programs for Miden VM is to use [Miden assembly](https://wiki.polygon.technology/docs/miden/user_docs/assembly/main).
+
+To get to know the language, check out our examples. It's better than writing in Circuits ...
+
+![](https://i.imgur.com/tyeMjD9.png)
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:35]: The right border of the drop down is cut out:

image


Which makes it look weird, it would be nice to make a new shot with the complete dropdown (and removing the editor from the background, that is unrelated to the point here)

File: playground/src/markdown/Explainer.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Want to know more? [Here](https://wiki.polygon.technology/docs/miden/user_docs/assembly/main).
+
+## Outputs of the Miden VM
+
+![](https://i.imgur.com/Bjgky7b.png)
+
+The `stack_output` represents our final state. The `overflow_addrs` is needed for verification. 
+
+The number of **cycles** just tells you how complex the computation is - it is the length of the execution trace. We need to always have an execution trace that is a power of 2 and >1024.
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:45]: Is the name `cycles` coming from the VM? This is a bit misleading and confusing, it should probably be `trace_len` or something like this, otherwise people will get confused why a 3 instruction program has so many cycles, and may result in people assuming this is super inefficient

File: playground/src/markdown/Explainer.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+The Outputs must also be a valid JSON (if you want to verify) and it can only contain numbers. 
+
+You can also test the VM by proving a program and tampering with the Outputs. See if you can still verify the set of (`stack_init`, `code`, `stack_output` and `overflow_addrs`)
+
+Want to know more? [Here](https://wiki.polygon.technology/docs/miden/user_docs/assembly/main).
+
+## OK, but what can I do now?
+
+![](https://i.imgur.com/Y6pkjzt.png)
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:55]: The way this render it seems the image is completely unrelated to the text:


image

It feels like a mistake. IMO it would be better to use the `hover` state of the button, which gives it a highlight, and add a screenshot for each section. I would also make the image smaller, we don't need to show the the UI in full size, just give context to the reader on what is being talked about.

File: playground/src/markdown/Explainer.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+### Run a program
+You can create a program and run it. There will be no proof generated which is much faster. Every program that successfully executes can also be proven, so I suggest using this functionality when hacking around. 
+
+### Debug a program
+You can step through the program and see the current VM state displayed in the Output section. 
+
+![](https://i.imgur.com/zmAGrVJ.png)
+
+* clk is the current (clock) cycle 
+* op is the current assembler operation (?)
```
Comment (hackaugusto) [inline playground/src/markdown/Explainer.md:66]: what are those question marks?

File: .github/workflows/gh-pages.yml [hunk 8]
```diff
@@ -3,7 +3,7 @@ name: Deploy Miden Assembly Playground
 on:
   push:
     branches:
-      - next
+      - dominik_add_explainer_page
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply .github/workflows/gh-pages.yml:6]: Jup that is only temporary such that I can already show the changes for my user interviews

https://0xpolygonmiden.github.io/examples/

File: playground/src/components/NavBar.tsx [hunk 9]
```diff
@@ -33,29 +33,39 @@ const NavBar = () => {
...
         </NavLink>
+        <NavLink
+          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
+          to="/examples/explainer/"
+          style={({ isActive }) => ({
+            color: isActive ? "#fff" : "#fff",
+            background: isActive ? "bg-blue-700" : "bg-blue-500",
+          })}
+        >
+          Ok ... What the f*ck?
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/src/components/NavBar.tsx:67]: Good point

File: playground/src/markdown/Explainer.md (hunk 5 above)
Comment (Dominik1999) [reply playground/src/markdown/Explainer.md:45]: this is a very nice idea, let me change that.

File: playground/src/markdown/Explainer.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/src/markdown/Explainer.md:1]: I think this is not supported by `react-markdown`. Normally `#` should create an anchor and and ID. 

Added some space for all the pages

File: playground/src/markdown/Explainer.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+## Inputs of the Miden VM
+External inputs can be provided to Miden VM in the Playground in two ways:
+
+![](https://i.imgur.com/n0LZpvU.png)
+
+* Public inputs - `stack_init` - can be supplied to the VM by initializing the stack with desired values before a program starts executing. Up to 16 stack items can be initialized in this way.
+* Secret (or nondeterministic) inputs - `advice_tape` - can be supplied to the VM. There is no limit on how much data the advice provider can hold. 
+
+*Check out the comparison example to see how the secret input works*
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/src/markdown/Explainer.md:22]: done

File: playground/src/markdown/Explainer.md (hunk 7 above)
Comment (Dominik1999) [reply playground/src/markdown/Explainer.md:66]: What do you mean?

File: playground/src/markdown/Explainer.md (hunk 6 above)
Comment (Dominik1999) [reply playground/src/markdown/Explainer.md:70]: I removed it

**Review by hackaugusto — APPROVED** [review]
LGTM, the text and layout will change later on, this is currently place holder
